e2echart/e2e-chart-template.html

```diff
@@ -405,43 +613,71 @@ <h5 class="modal-title">Resource</h5>
         for (const label in data) {
             const section = data[label]
             for (const sub in section) {
-                if (regex == null || (regex != null && regex.test(label))) {
-                    const data = section[sub];
-                    const totalDurationSeconds = data.reduce(
-                        (prev, curr) => prev + (curr.timeRange[1].getTime() - curr.timeRange[0].getTime())/1000,
-                        0);
+                const data = section[sub];
+                const totalDurationSeconds = data.reduce(
+                    (prev, curr) => prev + (curr.timeRange[1].getTime() - curr.timeRange[0].getTime())/1000,
+                    0);
 
-                    timelineData.push({label: label + sub + " " + getDurationString(totalDurationSeconds), data: data})
-                }
+                timelineData.push({label: label + sub + " " + getDurationString(totalDurationSeconds), data: data})
             }
         }
     }
 
-    function renderChart(regex) {
+    function renderChart() {
         var loc = window.location.href;
+        var ns = $('#namespace').val();
+        var category = $('#category').val();
+        var regexStr = $('#filterInput').val();
+        var regex = new RegExp(regexStr);
+        var filteredEvents;
+        // if none of the inputs are set, nothing to filter so don't waste time looping through everything
+        if (regexStr.length == 0 && ns.length == 0 && category.length == 0) {
+            filteredEvents = eventIntervals.items;
+        }
+        else {
+            // At least one of the inputs had a value, test any inputs that had values
+            // This currently does an OR operation of the input fields
+            filteredEvents = _.filter(eventIntervals.items, function(eventInterval) {
+                // Go ahead and filter out uncategorized events
+                if (eventInterval.categories.uncategorized) {
+                    return false;
+                }
+                var eventMatches = false;
+                if (regexStr.length > 0 && regex.test(eventInterval.locator)) {
+                    eventMatches = true;
+                }
+                if (ns.length > 0 && eventInterval.locatorObj.ns && eventInterval.locatorObj.ns.includes(ns)) {
+                    eventMatches = true;
+                }
+                if (category.length > 0 && eventInterval.categories[category]) {
+                    eventMatches = true;
+                }
+                return eventMatches;
+            });
+        }
+\
```

Comment:
this appears to cause the page to fail to render.  is it semantically important?